src/components/FilterByNumber.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useContext, useState } from 'react';
+import PlanetsContext from '../context/Planets';
+
+function FilterByNumber() {
+  const INITIAL_FILTER = {
+    column: 'population',
+    comparison: 'maior que',
+    value: '0',
+  };
+
+  const INITIAL_COLUMN_OPTIONS = [
+    'population',
+    'orbital_period',
+    'diameter',
+    'rotation_period',
+    'surface_water',
+  ];
+
+  const { handleFilterValues,
+    filterByNumericValues,
+    deleteNumericFilter,
+  } = useContext(PlanetsContext);
+
+  const [numericFilter, setNumericFilter] = useState(INITIAL_FILTER);
+  const [columnOptions, setColumnOptions] = useState(INITIAL_COLUMN_OPTIONS);
```

Comment:
regarding the requirements, looks like this initial state is useless

so u can use the same approach of single source of truth

```suggestion
  const INITIAL_COLUMN_OPTIONS = [
    'population',
    'orbital_period',
    'diameter',
    'rotation_period',
    'surface_water',
  ];

  const { handleFilterValues,
    filterByNumericValues,
    deleteNumericFilter,
  } = useContext(PlanetsContext);

  const [numericFilter, setNumericFilter] = useState(filterByNumericValues[0]);
  const [columnOptions, setColumnOptions] = useState(INITIAL_COLUMN_OPTIONS);
```